server/src/dbt_execution/DbtRpc.ts

```diff
@@ -54,14 +54,20 @@ export class DbtRpc extends Dbt {
       } catch (e) {
         this.finishWithError(e instanceof Error ? e.message : `Failed to start dbt-rpc. ${String(e)}`);
       }
-      this.doInitialCompile();
+      this.doInitialCompile().catch(e => console.log(`Error while compiling project. ${e instanceof Error ? e.message : String(e)}`));
     }
   }
 
-  doInitialCompile(): void {
-    this.dbtRpcClient.compile().catch(e => {
-      console.log(`Error while compiling project. ${e instanceof Error ? e.message : String(e)}`);
-    });
+  async doInitialCompile(): Promise<void> {
+    // Compilation can be started for a short time after the server receives a SIGHUP signal
+    await this.dbtRpcServer.ensureCompilationFinished();
```

Comment:
If it could be endless? Do we need some reasonable timeout for it?

Reply:
It usually takes several seconds. And since we don't wait initial compilation I think can do without the timeout for now.